include/secp256k1_silentpayments.h

```diff
@@ -0,0 +1,405 @@
+ *                            able to match the generated outputs to the
+ *                            correct silent payment addresses. The same
+ *                            recipient can be passed multiple times to create
+ *                            multiple outputs for the same recipient.
+ *              n_recipients: the number of recipients. This is equal to the
+ *                            total number of outputs to be generated as each
+ *                            recipient may passed multiple times to generate
+ *                            multiple outputs for the same recipient
+ *       outpoint_smallest36: serialized smallest outpoint (lexicographically)
+ *                            from the transaction inputs
+ *           taproot_seckeys: pointer to an array of pointers to 32-byte
+ *                            private keys of taproot inputs (can be NULL if no
+ *                            private keys of taproot inputs are used)
+ *         n_taproot_seckeys: the number of sender's taproot input private keys
+ *             plain_seckeys: pointer to an array of pointers to 32-byte
+ *                            private keys of non-taproot inputs (can be NULL
+ *                            if no private keys of non-taproot inputs are
+ *                            used)
+ *           n_plain_seckeys: the number of sender's non-taproot input private
+ *                            keys
+ */
+SECP256K1_API SECP256K1_WARN_UNUSED_RESULT int secp256k1_silentpayments_sender_create_outputs(
+    const secp256k1_context *ctx,
+    secp256k1_xonly_pubkey **generated_outputs,
+    const secp256k1_silentpayments_recipient **recipients,
+    size_t n_recipients,
+    const unsigned char *outpoint_smallest36,
+    const secp256k1_keypair * const *taproot_seckeys,
+    size_t n_taproot_seckeys,
+    const unsigned char * const *plain_seckeys,
+    size_t n_plain_seckeys
+) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3) SECP256K1_ARG_NONNULL(5);
+
+/** Create Silent Payment label tweak and label.
+ *
+ *  Given a recipient's scan key b_scan and a label integer m, calculate the
+ *  corresponding label tweak and label:
+ *
+ *  label_tweak = hash(b_scan || m) label = label_tweak * G
+ *
+ *  Returns: 1 if label tweak and label creation was successful.
+ *           0 if an error occured.
+ *  Args:                ctx: pointer to a context object
+ *  Out:               label: pointer to the resulting label public key
+ *             label_tweak32: pointer to the 32 byte label tweak
+ *  In:   recipient_scan_key: pointer to the recipient's scan key
+ *                         m: label integer (0 is used for change outputs)
+ */
+SECP256K1_API SECP256K1_WARN_UNUSED_RESULT int secp256k1_silentpayments_recipient_create_label_tweak(
+    const secp256k1_context *ctx,
+    secp256k1_pubkey *label,
+    unsigned char *label_tweak32,
+    const unsigned char *recipient_scan_key,
```

Comment:
nit: this change makes it clear that what is expected is the secret key

```suggestion
    const unsigned char *recipient_scan_seckey,
```